### DISCUSSION TITLE
Accessing user (auth) table info

### DISCUSSION THREAD

**Discussion by adrianjacob** [supabase/supabase#18722] [Questions]
Hi,

Am i right in thinking you cant query the auth.users table in the front end for security reasons?

E.g. i am creating a small app (3 tables) and 1 col in one table has a foreign key pointing to user.id (which returns fine) but I cant work out how to return the user.email as well?

  const { data, error } = await supabase
    .from("scores")
    .select("session, player, score, winner, sessions (game, date)");

returns

[
  {
    session: 1,
    player: '00c79606-ae0e-4046-87f9-131f87b4ccc8',
    score: 55,
    winner: true,
    sessions: { game: 342942, date: '2023-11-04' }
  },
  {
    session: 1,
    player: 'fa691ff5-ac04-4da1-9ca6-104c994206c3',
    score: 32,
    winner: false,
    sessions: { game: 342942, date: '2023-11-04' }
  }
]

But i want more (need) the user ID and their email.

Thanks

**Comment by GaryAustin1** [comment]
Normally this is solved by having a profiles table in the public schema set up by an auth.users trigger function.   Then you can join the profile info you want.
https://supabase.com/docs/guides/auth/managing-user-data#creating-user-tables
Read about profile table and then the triggers section further down.
